src/openrct2-ui/windows/Ride.cpp

```diff
@@ -6019,12 +6019,13 @@ static void window_ride_graphs_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi
         bottom = widget->bottom - widget->top - bottom - 13;
 
         // If red threshold is supported, adjust as well. Otherwise, keep 0.
-        if (!(redThresholdPositive == redThresholdNegative == 0))
+        if (redThresholdPositive != 0 && redThresholdNegative != 0)
```

Comment:
Discard my previous comment which showed wrong logic.
Because of left-associativity and unless I am wrong, `!(redThresholdPositive == redThresholdNegative == 0)` resolves the same as `!((redThresholdPositive == redThresholdNegative) == 0)`, ie `redThresholdPositive == redThresholdNegative`.

Do not mix up comparison and definition: `a = b = 0` does what you intuitively think, `a == b == 0` does *not*.
For this reason, such chained constructions shall be avoided at all costs.